Tests/EndToEnd.Tests/Tests/CheckSubtaskingTreeUnifiedApi/SubtaskingTreeUnifiedApiWorker.cs

```diff
@@ -0,0 +1,189 @@
+// This file is part of the ArmoniK project
+// 
+// Copyright (C) ANEO, 2021-2022. All rights reserved.
+//   W. Kirschenmann   <wkirschenmann@aneo.fr>
+//   J. Gurhem         <jgurhem@aneo.fr>
+//   D. Dubuc          <ddubuc@aneo.fr>
+//   L. Ziane Khodja   <lzianekhodja@aneo.fr>
+//   F. Lemaitre       <flemaitre@aneo.fr>
+//   S. Djebbar        <sdjebbar@aneo.fr>
+// 
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published
+// by the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+// 
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+// 
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+using ArmoniK.DevelopmentKit.Common;
+using ArmoniK.DevelopmentKit.Common.Exceptions;
+using ArmoniK.DevelopmentKit.Worker.Grid;
+using ArmoniK.EndToEndTests.Common;
+
+using Microsoft.Extensions.Logging;
+
+namespace ArmoniK.EndToEndTests.Tests.CheckSubtaskingTreeUnifiedApi;
+
+public class SubtaskingTreeUnifiedApiWorker : TaskSubmitterWorkerService
+{
+  public void CheckPayload(ClientPayload payload)
+  {
+    if (payload == null)
+    {
+      throw new ArgumentException("CheckPayload: payload must not be null");
+    }
+
+    if (payload.Type != ClientPayload.TaskType.Aggregation)
+    {
+      if (payload.Numbers == null)
+      {
+        throw new ArgumentException("CheckPayload: payload numbers must not be null");
+      }
+
+      if (payload.NbSubTasks <= 1)
+      {
+        throw new ArgumentException($"CheckPayload: payload.NbSubTasks:{payload.NbSubTasks} must be >= 2");
+      }
+    }
+  }
+
+  private static IEnumerable<T[]> SplitList<T>(List<T> listToSplit,
+                                               int     nbSplit = 2)
+    => listToSplit.Chunk((int)Math.Ceiling(listToSplit.Count / (decimal)nbSplit));
```

Comment:
```suggestion
    => listToSplit.Chunk((int)Math.Ceiling(listToSplit.Count / (float)nbSplit));
```
No need for a decimal, a bit too heavy I think